--- a/kernel-module-imx-gpu-viv-src/Kbuild	2026-07-11 15:22:27.659550678 -0500
+++ b/kernel-module-imx-gpu-viv-src/Kbuild	2026-07-11 15:22:27.668570281 -0500
@@ -335,6 +335,8 @@
 
 $(MODULE_NAME)-objs  = $(OBJS)
 
+EXTRA_CFLAGS += -Wno-error
+
 ccflags-y += $(EXTRA_CFLAGS)
 subdir-ccflags-y += $(EXTRA_CFLAGS)
 
--- a/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/gc_hal_kernel_linux.h	2026-07-11 15:22:27.661330802 -0500
+++ b/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/gc_hal_kernel_linux.h	2026-07-11 15:22:27.668795526 -0500
@@ -383,4 +383,8 @@
 gceSTATUS
 gckIOMMU_Construct(IN gckOS Os, OUT gckIOMMU *Iommu);
 
+#ifndef in_irq
+#define in_irq() in_hardirq()
+#endif
+
 #endif /* __gc_hal_kernel_linux_h_ */

--- a/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c	2026-07-11 15:22:27.664320180 -0500
+++ b/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dma.c	2026-07-11 15:22:27.670133426 -0500
@@ -60,6 +60,7 @@
 #include <linux/seq_file.h>
 #include <linux/mman.h>
 #include <linux/atomic.h>
+#include <linux/mm.h>
 #include <linux/dma-mapping.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0)
 #    include <linux/dma-direct.h>
--- a/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c	2026-07-11 15:22:27.666417587 -0500
+++ b/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c	2026-07-11 15:22:27.670397812 -0500
@@ -61,6 +61,10 @@
 #include <linux/atomic.h>
 #include <linux/dma-mapping.h>
 #include <linux/slab.h>
+#include <linux/mm.h>
+#ifndef MAX_ORDER
+#define MAX_ORDER MAX_PAGE_ORDER
+#endif
 #if defined(CONFIG_X86) && (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0))
 #    include <asm/set_memory.h>
 #endif
--- a/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/gc_hal_kernel_driver.c	2026-07-11 15:22:27.668100283 -0500
+++ b/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/gc_hal_kernel_driver.c	2026-07-11 15:22:27.670659734 -0500
@@ -70,7 +70,7 @@
 MODULE_DESCRIPTION("Vivante Graphics Driver");
 MODULE_LICENSE("Dual MIT/GPL");
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
-MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+/* MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver); */
 #endif
 
 static struct class *gpuClass;
@@ -1149,7 +1149,9 @@
     return ret;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 11, 0)
+static void viv_dev_remove(struct platform_device *pdev)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
 static int viv_dev_remove(struct platform_device *pdev)
 #else
 static int __devexit viv_dev_remove(struct platform_device *pdev)
@@ -1177,7 +1179,11 @@
     }
 
     gcmkFOOTER_NO();
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 11, 0)
+    return;
+#else
     return 0;
+#endif
 }
 
 static void viv_dev_shutdown(struct platform_device *pdev)
