diff -urN '--exclude=*.o' '--exclude=*.ko' '--exclude=*.cmd' '--exclude=*.mod' '--exclude=*.mod.c' '--exclude=modules.order' '--exclude=Module.symvers' '--exclude=.tmp_versions' aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/aicwf_sdio.c aic8800-driver-master/drivers/aic8800/aic8800_fdrv/aicwf_sdio.c
--- aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/aicwf_sdio.c	2026-06-25 06:55:41.000000000 -0500
+++ aic8800-driver-master/drivers/aic8800/aic8800_fdrv/aicwf_sdio.c	2026-07-12 11:45:42.374942497 -0500
@@ -527,7 +527,7 @@
 
     aicwf_sdio_pwrctl_timer(sdiodev, 0);
     if(timer_pending(&sdiodev->rwnx_hw->p2p_alive_timer)){
-        ret = del_timer(&sdiodev->rwnx_hw->p2p_alive_timer);}
+        ret = timer_delete(&sdiodev->rwnx_hw->p2p_alive_timer);}
     sdio_dbg("%s\n",__func__);
     if (sdiodev->pwrctl_tsk) {
         complete(&sdiodev->pwrctrl_trgg);
@@ -1019,7 +1019,7 @@
     unsigned long flags = 0;
 
     spin_lock_irqsave(&rx_priv->rxqlock, flags);
-    if (!aicwf_rxframe_enqueue(sdiodev->dev, &rx_priv->rxq, pkt)) {
+    if (!aicwf_rxframe_enqueue(sdiodev->dev, (struct frame_queue *)&rx_priv->rxq, pkt)) {
         spin_unlock_irqrestore(&rx_priv->rxqlock, flags);
         aicwf_dev_skb_free(pkt);
         return;
@@ -1082,7 +1082,7 @@
     spin_lock_bh(&sdiodev->pwrctl_lock);
     if (!duration) {
         if (timer_pending(&sdiodev->timer))
-            del_timer_sync(&sdiodev->timer);
+            timer_delete_sync(&sdiodev->timer);
     } else {
         sdiodev->active_duration = duration;
         timeout = msecs_to_jiffies(sdiodev->active_duration);
diff -urN '--exclude=*.o' '--exclude=*.ko' '--exclude=*.cmd' '--exclude=*.mod' '--exclude=*.mod.c' '--exclude=modules.order' '--exclude=Module.symvers' '--exclude=.tmp_versions' aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/aicwf_txrxif.c aic8800-driver-master/drivers/aic8800/aic8800_fdrv/aicwf_txrxif.c
--- aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/aicwf_txrxif.c	2026-06-25 06:55:41.000000000 -0500
+++ aic8800-driver-master/drivers/aic8800/aic8800_fdrv/aicwf_txrxif.c	2026-07-12 11:45:42.375283681 -0500
@@ -323,6 +323,23 @@
     kfree(tx_priv);
 }
 
+static bool aicwf_another_ptk(struct sk_buff *skb)
+{
+	u8 *data;
+	u16 aggr_len = 0;
+
+	if (skb->data == NULL || skb->len < 2) {
+		return false;
+	}
+	data = skb->data;
+	aggr_len = (*skb->data | (*(skb->data + 1) << 8));
+	if (aggr_len == 0) {
+		return false;
+	}
+
+	return true;
+}
+
 #ifdef CONFIG_PREALLOC_RX_SKB
 static bool aicwf_another_ptk_1(struct rx_buff *buffer)
 {
@@ -344,23 +361,6 @@
 
     return true;
 }
-#else
-static bool aicwf_another_ptk(struct sk_buff *skb)
-{
-	u8 *data;
-	u16 aggr_len = 0;
-
-	if (skb->data == NULL || skb->len < 2) {
-		return false;
-	}
-	data = skb->data;
-	aggr_len = (*skb->data | (*(skb->data + 1) << 8));
-	if (aggr_len == 0) {
-		return false;
-	}
-
-	return true;
-}
 #endif
 
 
@@ -477,11 +477,11 @@
 
     while (1) {
         spin_lock_irqsave(&rx_priv->rxqlock, flags);
-        if(aicwf_is_framequeue_empty(&rx_priv->rxq)) {
+        if(aicwf_is_framequeue_empty((struct frame_queue *)&rx_priv->rxq)) {
             spin_unlock_irqrestore(&rx_priv->rxqlock,flags);
             break;
         }
-        skb = aicwf_frame_dequeue(&rx_priv->rxq);
+        skb = aicwf_frame_dequeue((struct frame_queue *)&rx_priv->rxq);
         spin_unlock_irqrestore(&rx_priv->rxqlock,flags);
         if (skb == NULL) {
             txrx_err("skb_error\r\n");
diff -urN '--exclude=*.o' '--exclude=*.ko' '--exclude=*.cmd' '--exclude=*.mod' '--exclude=*.mod.c' '--exclude=modules.order' '--exclude=Module.symvers' '--exclude=.tmp_versions' aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/rwnx_cmds.c aic8800-driver-master/drivers/aic8800/aic8800_fdrv/rwnx_cmds.c
--- aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/rwnx_cmds.c	2026-06-25 06:55:41.000000000 -0500
+++ aic8800-driver-master/drivers/aic8800/aic8800_fdrv/rwnx_cmds.c	2026-07-12 11:45:42.375800087 -0500
@@ -444,7 +444,7 @@
 #endif
 #ifdef AICWF_USB_SUPPORT
     struct aic_usb_dev *usbdev = container_of(cmd_mgr, struct aic_usb_dev, cmd_mgr);
-    struct rwnx_hw *rwnx_hw = usbdev->rwnx_hw;
+    struct rwnx_hw *rwnx_hw_usb = usbdev->rwnx_hw; (void)rwnx_hw_usb;
 #endif
     struct rwnx_cmd *cmd, *pos;
     bool found = false;
diff -urN '--exclude=*.o' '--exclude=*.ko' '--exclude=*.cmd' '--exclude=*.mod' '--exclude=*.mod.c' '--exclude=modules.order' '--exclude=Module.symvers' '--exclude=.tmp_versions' aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800-driver-master/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800-driver-master.orig/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2026-06-25 06:55:41.000000000 -0500
+++ aic8800-driver-master/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2026-07-12 11:45:42.377070563 -0500
@@ -1722,7 +1722,7 @@
             break;
         case (SIOCDEVPRIVATE+1):
             AICWFDBG(LOGINFO, "IOCTL PRIVATE\n");
-            ret = android_priv_cmd(net, req, cmd);
+            ret = -EOPNOTSUPP;
 	    break;
         default:
             ret = -EOPNOTSUPP;
